src/lib/plugin-loader.js

```diff
@@ -31,47 +32,71 @@ function try_load(path) {
  */
 function load_plugins(config, plugin_configs, params, sanity_check) {
   let plugins = Object.keys(plugin_configs || {}).map(function(p) {
-    let plugin;
 
-    // try local plugins first
-    plugin = try_load(Path.resolve(__dirname + '/..//plugins', p));
+    let resolver = function(resolve, reject) {
+      let plugin;
+
+      // try local plugins first
+      plugin = try_load(Path.resolve(__dirname + '/..//plugins', p));
 
-    // npm package
-    if (plugin === null && p.match(/^[^\.\/]/)) {
-      plugin = try_load(`verdaccio-${p}`);
-      // compatibility for old sinopia plugins
-      if (!plugin) {
-        plugin = try_load(`sinopia-${p}`);
+      // npm package
+      if (plugin === null && p.match(/^[^\.\/]/)) {
+        plugin = try_load(`verdaccio-${p}`);
+        // compatibility for old sinopia plugins
+        if (!plugin) {
+          plugin = try_load(`sinopia-${p}`);
+        }
       }
-    }
 
-    if (plugin === null) {
-      plugin = try_load(p);
-    }
+      if (plugin === null) {
+        plugin = try_load(p);
+      }
 
-    // relative to config path
-    if (plugin === null && p.match(/^\.\.?($|\/)/)) {
-      plugin = try_load(Path.resolve(Path.dirname(config.self_path), p));
-    }
+      // relative to config path
+      if (plugin === null && p.match(/^\.\.?($|\/)/)) {
+        plugin = try_load(Path.resolve(Path.dirname(config.self_path), p));
+      }
 
-    if (plugin === null) {
-      throw Error('"' + p + '" plugin not found\ntry "npm install verdaccio-' + p + '"');
-    }
+      if (plugin === null) {
+        reject('' + p + ' plugin not found. try "npm install verdaccio-' + p + '"');
+      }
 
-    if (typeof(plugin) !== 'function') {
```

Comment:
Answering your question, I am afraid if we don't force to stop the app we might allow a usage of corrupted plugins. I guess if one plugin does not work we have to be strict, otherwise, it will lead us to wrong behaviors that people won't understand. We need to find a better solution